scrapy/crawler.py

```diff
@@ -465,66 +490,49 @@ def crawl(
                 "it must be a spider class (or a Crawler object)"
             )
         if not is_asyncio_reactor_installed():
-            raise RuntimeError("AsyncCrawlerRunner requires AsyncioSelectorReactor.")
+            raise RuntimeError(
+                f"{type(self).__name__} requires AsyncioSelectorReactor."
+            )
         crawler = self.create_crawler(crawler_or_spidercls)
         return self._crawl(crawler, *args, **kwargs)
 
-    def _crawl(
-        self, crawler: Crawler, *args: Any, **kwargs: Any
-    ) -> asyncio.Future[None]:
+    def _crawl(self, crawler: Crawler, *args: Any, **kwargs: Any) -> asyncio.Task[None]:
+        # At this point the asyncio loop has been installed either by the user
+        # or by AsyncCrawlerProcess (but it isn't running yet, so no asyncio.create_task()).
+        loop = asyncio.get_event_loop()
         self.crawlers.add(crawler)
-        future = deferred_to_future(crawler.crawl(*args, **kwargs))
-        self._active.add(future)
+        task = loop.create_task(crawler.crawl_async(*args, **kwargs))
+        self._active.add(task)
 
-        def _done(_: asyncio.Future[None]) -> None:
+        def _done(_: asyncio.Task[None]) -> None:
             self.crawlers.discard(crawler)
-            self._active.discard(future)
+            self._active.discard(task)
             self.bootstrap_failed |= not getattr(crawler, "spider", None)
 
-        future.add_done_callback(_done)
-        return future
+        task.add_done_callback(_done)
+        return task
 
     async def stop(self) -> None:
         """
         Stops simultaneously all the crawling jobs taking place.
 
         Completes when they all have ended.
         """
-        await deferred_to_future(self._stop())
+        if self.crawlers:
+            await asyncio.wait(
+                [asyncio.create_task(c.stop_async()) for c in self.crawlers]
+            )
 
     async def join(self) -> None:
         """
         Completes when all managed :attr:`crawlers` have completed their
         executions.
         """
         while self._active:
-            await asyncio.gather(*self._active)
-
-
-class CrawlerProcess(CrawlerRunner):
-    """
-    A class to run multiple scrapy crawlers in a process simultaneously.
+            await asyncio.wait(self._active)
```

Comment:
`asyncio.gather()` has different exception handling semantics wrt `yield DeferredList()` (the former raises the first exception, the latter simply records it in the result), `asyncio.wait()` in this case does what we want, waiting until all tasks finish in some way.